pyramid_swagger/renderer.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+This model contains the main factory of renderers to use while dealing with Swagger 2.0 endpoints.
+"""
+from functools import partial
+
+from bravado_core.exception import MatchingResponseNotFound
+from bravado_core.exception import SwaggerMappingError
+from bravado_core.marshal import marshal_schema_object
+from bravado_core.response import get_response_spec
+from pyramid.renderers import JSON
+
+
+class PyramidSwaggerRendererFactory(object):
+    def __init__(self, renderer_factory=JSON()):
+        self.renderer_factory = renderer_factory
+
+    def _marshal_object(self, request, response_object):
+        # operation attribute is injected by validator_tween in case the endpoint is served by Swagger 2.0 specs
+        operation = getattr(request, 'operation')
+
+        if not operation:
+            # If the request is not served by Swagger2.0 endpoint _marshal_object is NO_OP
+            return response_object
+
+        try:
+            response_spec = get_response_spec(
+                status_code=request.response.status_code,
+                op=request.operation,
+            )
+            return marshal_schema_object(
+                swagger_spec=request.registry.settings['pyramid_swagger.schema20'],
+                schema_object_spec=response_spec['schema'],
+                value=response_object,
+            )
+        except MatchingResponseNotFound:
+            # Response specs not found
+            return response_object
+        except SwaggerMappingError:
```

Comment:
Under what circumstances will this occur? Is silently no-oping the right response?

Reply:
``SwaggerMappingError`` could be raised if (the list is not exhaustive but gives an idea about the implications)
 - the response spec is not specifying the type (which is weird but possible, especially if no response validation is defined) -> https://github.com/Yelp/bravado-core/blob/619dec76ecbb67858129b808e8af4e5f8496e7a6/bravado_core/marshal.py#L60
 - the response body is not conform to the response spec (attribute ``attr`` is specified to be of type ``object``, but the value to marshal has ``attr`` set to ``1``)  -> https://github.com/Yelp/bravado-core/blob/619dec76ecbb67858129b808e8af4e5f8496e7a6/bravado_core/marshal.py#L133
 - An ``attr`` is defined  as type ``string`` and format ``date`` but the view returned object has ``attr`` set to date string representation and not ``datetime.date`` instance -> https://github.com/Yelp/bravado-core/blob/619dec76ecbb67858129b808e8af4e5f8496e7a6/bravado_core/formatter.py#L43

The idea of acting as NO OP is because this cases should be already handled by the downstream renderer or the service is returning a not spec compliant response. (which is the current behaviour)

We can eventually make sure that this conditions are made more visible maybe with WARNING logs ... what do you think?

Reply:
Hm fair point on not wanting to change current behavior. Maybe let's keep an eye out for whether this needs logging as we go forward? No need to add it now, considering we don't log anywhere else yet :<